### ISSUE TITLE
Open ssl Version 

### ISSUE THREAD

**Issue by HumbleLearner** [erlang/otp#6499]
**Describe the bug**
I use Rabbitmq for my product which in turn uses Erlang. When I scan the product using black duck scan for security vulnerability, it indicates Erlang 25.1.2 is using Crypto.dll (5.2.1) which is using a vulnerable Openssl 1.1.1q. 
Is this vulnerability valid ?
Is crypto.dll using OpenSll.1.1.1q or some other latest version?

Any information on this will be helpful

**To Reproduce**
Security vulnerability scan

**Expected behavior**
It should be using latest version of openssl as the old version is having vulnerability 

**Affected versions**
25.1.2or lower

**Comment by IngelaAndin** [comment]
As I answered in Erlang forums OTP-25.1.2 does not depend on a special version of OpenSSL, that depends on your build. If the vulnerability is for OpenSSL protocol implementation it is not relevant for OTP only if the vulnerability is on the libcrypto functions will it also affect OTP.

You can use crypto:info_lib to/0  in an erlang shell to see what version of OpenSSL's cryptolib is compiled and linked to be used by OTP.

**Comment by HumbleLearner** [comment]
Thanks for the quick reply Ingela, I get below error when I try to run that command in the shell
crypto:info_lib to/0.
* 1:17: syntax error before: to

**Comment by IngelaAndin** [comment]
It should be called like this: 

```
Erlang/OTP 25 [erts-13.1.2] [source-e030d309a1] [64-bit] [smp:16:16] [ds:16:16:10] [async-threads:1] [jit:ns]

Eshell V13.1.2  (abort with ^G)
1>  crypto:info_lib().
[{<<"OpenSSL">>,269488239,
  <<"OpenSSL 1.1.1f  31 Mar 2020">>}]
2> 

```

The syntax used above is how we refer to a function specifying its arity. The same function names with a different number of arguments are different functions.

**Comment by IngelaAndin** [comment]
Note, I updated my comment as it seems a mention of an irrelevant use of "to" also snook in!

**Comment by HumbleLearner** [comment]
So what I understand, if there is vulnerability in libcrypto  function than it will effect OTP. 
When I run the command, this is the output I get :-
crypto:info_lib(). 
[{<<"OpenSSL">>,269488207,
  <<"OpenSSL 1.1.1d  10 Sep 2019">>}]

I am using Erlang build 25.1.2
Is there a way to update Openssl to 1.1.1q?

**Comment by IngelaAndin** [comment]
You can use the following configure flag 

`--with-ssl=PATH - Specify base location of OpenSSL include and lib directories`  

while building OTP. See also 

https://www.erlang.org/doc/installation_guide/install  (which however seems to include a lot of legacy flags for OpenSSL that are not relevant).

**Comment by IngelaAndin** [comment]
Oh, by the way, are you by any chance running windows? And which is the CVE you are worried about?

**Comment by HumbleLearner** [comment]
Yes I am using windows, there are series of vulnerability which are related to OpenSSL 1.1.1d such as (CVE-2022-1292, CVE-2021-3711, CVE-2022-2068, CVE-2022-0778, 
CVE-2020-1967, CVE-2021-23840, CVE-2021-3712) so it may make product vulnerable.

**Comment by IngelaAndin** [comment]
Most of these CVS are totally irrelevant to OTP. But I will investigate one of them a little more and get back to you.

**Comment by IngelaAndin** [comment]
This CVE-2021-23840 is relevant for users of the function crypto:crypto_one_time_aead/6
or crypto:crypto_one_time_aead/7.  If the InText argument is very close to 2 Gbyte. 
Due to protocol limitations, this should not happen in the Erlang SSL (TLS) application. But another user of crypto could be vulnerable. We will update the OpenSSL version of our windows build.

**Comment by HumbleLearner** [comment]
Thanks Ingela, If you update then by when we will be able to download the latest build.

**Comment by proxyles** [comment]
We have updated our builds and next official versions will include the update. 
Next planned release is OTP 25.2 14/12-22

**Comment by HumbleLearner** [comment]
Thanks for resolving this.
